### ISSUE TITLE
difference between thv and thvm?

### ISSUE THREAD

**Issue by adamrher** [adamrher/CAM#2]
In clubb_intr.F90,

       thv(i,k) = state1%t(i,k)*inv_exner_clubb(i,k)*(1._r8+zvir*state1%q(i,k,ixq)&
                  -state1%q(i,k,ixcldliq))

It subtracts cldliq? I don't know this form. However, thv_ds_zt is set to thv as computed above. 

In advance_clubb_core_module, there is a calculation:

      thvm = thlm + ep1 * thv_ds_zt * rtm &
                  + ( Lv/(Cp*exner) - ep2 * thv_ds_zt ) * rcm

And in the initial clubb+mf implementation, this thvm is used for (among other things) the calculation of buoyancy as the reference buoyancy.

It turns out using these different thv calculations leads to large differences in the mass fluxes ... With using the former (i.e., thv_ds_zt), the mass fluxes are larger.

@mikaelwitte do you have any incite into the differences here?

**Comment by adamrher** [comment]
After looking at this for a while. I think the reason that thvm is backed out from thlm in advance_clubb_core is b/c the definition of thv = f(T), but T isn't known. Only thl is known, as it's the prognostic variable.

thl is actually computed in clubb_intr using T, to initialize clubb. So I'm thinking of calculation thv = f(T). That this gives drastically different results from backing out of thv from thl is a problem, but I will let that be a CLUBB problem for now, and keep this jumbling out of clubb_intr.

**Comment by adamrher** [comment]
re: subtracting ql in calculating thv. Peter L and I were looking at the dycore side and saw that, at least for Tv, there is no subraction of ql. [The AMS def of Tv says](https://glossary.ametsoc.org/wiki/Virtual_temperature) "Some authors incorporate the density increment due to liquid or solid water into virtual temperature," i.e., the subtraction of ql. So I'm inclined to remove this ql correction from thv, but will probably check with Peter Bogenschutz first to understand his reasoning for computing it that way in clubb_intr.F90

**Comment by mikaelwitte** [comment]
Perhaps for dycore purposes the ql contribution isn't important, but I think it's considerably more important for moist convection. I would strongly recommend comparing output with both formulations for a deep convective case before making a final judgment.

**Comment by adamrher** [comment]
OK. Glad to hear you have an opinion on the physics of this. I'm solely coming at this from "is it computed consistently throughout the model" due to my lack of knowledge on this formula. I haven't committed any changes discussed in this thread to a branch yet. If I do, I'll make sure to include before/after plots and justification in this thread.

**Comment by bogensch** [comment]
hmmm, I'm not sure I recall the reasoning for using that definition.  In advance_clubb_core I see CLUBB expects the "dry theta_v" which to me seems to suggest it wants the theta_v definition without the condensate loading.  I would ping Vince to get his opinion on whether this is an inconsistency between the CLUBB interface and CLUBB.

**Comment by adamrher** [comment]
That's a good point Pete ... there are all these variables in clubb_intr like:

   real(r8) :: thv_ds_zm(pverp+1-top_lev)                       ! Dry, base-state theta_v on momentum levels

Logic would have you think the v in theta_v indicates it's not dry ... so maybe dry refers to non-gaseous impacts on theta_v, like condensate loading as you suggest ...

**Comment by mikaelwitte** [comment]
I was chatting with Rachel Storer (@rls347) about this since she's also done a good amount of work with CLUBB in the past. She was as confused as us and rearranged Vince's thvm equation - it's basically identical to the thv calculation with an extra Exner term, so maybe that's the difference you're seeing in the SCAM output? If Exner is applied twice, I can see thvm would be higher and MF plumes would be "stronger," so to speak.

If dry thv (let's go with dry="no condensate loading" as Pete suggests) is what's used in the dycore, then the thvm calculation would still add condensate loading, and I don't see any reason to remove this. It looks to me like we should be OK if we can sort out what's going on with the extra Exner function that Rachel found - either that, or just switch to using dycore "dry" thv as an input.

**Comment by rls347** [comment]
It also uses the 'mean state' theta_v rather than theta or theta_L for part of it. 
thv = theta + thvs ( .61qv - qc) 
I'm not sure why would this be desirable but maybe it was just reducing the computational cost of computing theta before adjusting for moisture. 

The exner that I wasn't sure about was in the theta_L --> theta part of the calculation

**Comment by adamrher** [comment]
@rls347 thanks for working this out! Not to sound like a school teacher, but could you show ur work using the CLUBB variable names (e.g., thvm, ep1, rcm, ...) in a post here? I would like to have a record of this so we can show it to Vince for feedback.

**Comment by rls347** [comment]
![oT5fFjYJSau1q81ivEoR%g](https://user-images.githubusercontent.com/12312018/95145944-5b362880-0731-11eb-8b9e-c1b3807844ef.jpg)

**Comment by adamrher** [comment]
Thanks Rachel. This result:

theta_v = theta + theta_vds (0.61*qv - ql)

Only collapses into the standard definition of theta_v, if theta = theta_vds:

theta_v = theta + theta(0.61qv - ql)

My takeaway is that I really need to figure out what the proper definition of theta_vds is supposed to be.

**Comment by rls347** [comment]
theta_v_ds is the theta_v that's calculated outside of CLUBB and passed in. I'm not sure why that's used for this rather than the theta that was just calculated. There would be a .61qv difference there unless I'm missing some other reasoning. 

I'm also not sure still why the Exner function is in there, so it's definitely possible I'm missing something.

**Comment by adamrher** [comment]
ahh, so one reason thvm ~= theta_v_ds may be because thvm is calculated after the pdf_closure_driver in advance_clubb_core, which would change the values of rcm and rvm = rtm-rcm. That kind of makes sense, no?

**Comment by mikaelwitte** [comment]
I think the Exner term is just due to the definition of theta_l. Further up in clubb_intr, mean theta_l is defined as thlm(i,k) = ( state1%t(i,k) - (latvap/cpairv(i,k,lchnk))*state1%q(i,k,ixcldliq) )   * inv_exner_clubb(i,k) = theta - Lv/(cp*Exner)ql. This checks out to me.

The big question I still have is what "dry state" means - Adam, I think you were onto something earlier when you suggested that it may just neglect condensate loading, so this could be a mechanism to account for the case where CAM "sees" q=qt as opposed to q being partitioned between vapor and condensate. In other words, maybe defining a "dry state" theta_v is irrelevant now that liquid water is prognostic?

And just saw your last comment - that would make sense too

**Comment by adamrher** [comment]
Mikael, If I understand you correctly, I think this make sense. rcm doesn't appear to be initialized in clubb, and so presumably zero before the pdf_closure_driver. And so it follows that theta_v_ds probably should not have any assumption about liquid in it.

**Comment by mikaelwitte** [comment]
That seems reasonable - it would still probably be good to check with Vince that our interpretation is correct though.

To recap:
- Base state theta_v (calculated in clubb_intr as variable thv, passed in/out of advance_clubb_core as thv_ds_zt) is initialized from the input CAM state with prognostic condensate amount q(...ixcldliq). **Note**: this does NOT include cloud ice.
- thvm is calculated internally every *t*th substep (i.e. t=1,nadv) in advance_clubb_core using the "base state" thv_ds_dt and the updated diagnostic partitioning of moisture between vapor and condensate from CLUBB's PDF closure. 
- At the end of clubb_intr, rcm is output to q(...ixcldliq) and condensate is further modified by microphysics, leading to a new base state the next time clubb_tend_cam is called.
- Fun little twist: the value of thv calculated in the PBL depth diagnosis at the end of clubb_tend_cam reverts to ignoring condensate loading - but we can leave that inconsistency for another time

**Comment by rls347** [comment]
@adamrher that could make sense, if the adjustment is so that the theta that's used is closer to the initial state.

So I don't think the 'base state' value ignores all condensate loading, since it's subtracted off in clubb_intr, but it would then ignore the condensate loading generated in that timestep. Or, some but not all of it? Since presumably
thetav_ds(.61qv-qc) > theta(.61qv-qc)

**Comment by rls347** [comment]
Okay..if state1%t is theta, and not T, then yes, the Exner function is consistent there. Sigh model variables.

**Comment by adamrher** [comment]
yeah, thv_ds does subtract cloud liquid. I think from our discussion that it probably shouldn't. I'm going to fire off an email to Vince tomorrow after I meet with Peter to get a few more clarifications. I'll cc yall. The relevant workflow in clubb_intr is as follows:

rcm_inout <- this is actually taken from pbuf, and so I need to debug to understand whether this is preserved every time-step. Might be better to use this for thv -> integrate_mf if it is preserved per time-step.

rtm_in = state1%q(i,k,ixq) + state1%q(i,k,ixcldliq)
thv_ds = T/exner * (1 + zvir*state1%q(i,k,ixq) - state1%q(i,k,ixcldliq))

t = 1,nadv
   integrate_mf(which thv?) <--it's too bad I moved this out of clubb actually ... prefer to be in sync with rcm from pdf_closure
   call clubb (theta_v_ds)
   ->inside clubb
       compute rcm_inout from pdf_clousre
       thvm = th + thv_ds*(zvir*(rtm-rcm) - rcm) <-Rachel's simplification
end do

ptend_loc%q(i,k,ixq) = (rtm(i,k)-rcm(i,k)-state1%q(i,k,ixq))/hdtime
ptend_loc%q(i,k,ixcldliq) = (rcm(i,k)-state1%q(i,k,ixcldliq))/hdtime
update state

Then as Mikael says, PBL, ustar and m-o are fed in a new thv:
thv(i,k) = th(i,k)*(1.0_r8+zvir*state1%q(i,k,ixq))

**Comment by mikaelwitte** [comment]
I was just about to submit another comment pointing out that rcm comes from pbuf so is not necessarily 0 - you beat me to it.

I'm still not sure why you think we should get rid of the liquid contribution. CAM has prognostic condensate so there's no inconsistency to include this in the "base state," and condensate loading is crucial for correctly calculating buoyancy in a cloudy atmosphere. I am beginning to more strongly suspect that the "ds" part of the variable name is a holdover from an older version of CAM with diagnostic ql.

**Comment by adamrher** [comment]
I think the answer depends on what clubb thinks thv_ds is, in deciding whether to Include or not include condensate loading. 

update, rcm is preserved from time-step to time-step.

**Comment by rls347** [comment]
I agree with Mikael. Condensate loading is important physics and needs to be included.

Vince does still refer to a 'dry base state' theta_v in his CLUBB documentation. But in pdf_closure, the calculation of theta_v is even noted as "an approximated and linearized form of the theta_v equation". So it seems like CLUBB's calculation is as it is for computational or simplification reasons rather than physical ones.
